src/ctia/bundle/core.clj

```diff
@@ -162,14 +203,26 @@
 (s/defn with-existing-entity :- EntityImportData
   "If the entity has already been imported, update the import data
    with its ID. If more than one old entity is linked to an external id,
-   an error is reported."
+   an error is reported. If realized :id is provided, asserts that entity
+   must exist, otherwise errors."
   [{:keys [external_ids]
     :as entity-data} :- EntityImportData
    find-by-external-id :- (s/=> s/Any (s/named s/Any 'external_id))
+   id->old-entity :- {s/Str (s/pred map?)}
    services :- HTTPShowServices]
-  (if-let [old-entities (mapcat find-by-external-id external_ids)]
-    (let [old-entity (some-> old-entities
-                             first
+  (if-some [realized-id (when-some [new-id (get-in entity-data [:new-entity :id])]
+                          (when-not (schemas/transient-id? new-id)
+                            new-id))]
+    (if-some [old-entity (id->old-entity realized-id)]
+      (assoc entity-data :result "exists" :id (:id old-entity))
+      (cond-> entity-data
+        ;; if we want to support specifying ids on creation, start here
+        true ;;(not (auth/capable? auth-identity :specify-id))
+        (assoc :error {:type :unresolvable-id
+                       :reason (str "Long id must already correspond to an entity: " realized-id)}
+               :result "error")))
```

Comment:
you will have to combine this behaviour with the update entity option, right?
At some point we might need to split this function with the 2 branches of the `if-some` to keep it "easy" to read.

Reply:
The update existing logic was moved to `prepare-bulk`, where I now return a separate result `:unsubmitted-result` containing both errors and existing (when appropriate).